internal/api/connectors/sources/impl.go

```diff
@@ -0,0 +1,146 @@
+package sources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"playbook-dispatcher/internal/common/constants"
+	"playbook-dispatcher/internal/common/utils"
+	"time"
+
+	"github.com/redhatinsights/platform-go-middlewares/request_id"
+	"github.com/spf13/viper"
+)
+
+const (
+	basePath               = "/api/sources/v3.1/"
+	filterPath QueryFilter = "filter[source_ref][eq]="
+)
+
+type SourcesClient interface {
+	GetSources(
+		ctx context.Context,
+		sourceId ID,
+	) (sources []SourceData, err error)
+
+	GetRHCConnectionStatus(
+		ctx context.Context,
+		sourceId ID,
+	) (status RhcConnectionStatus, err error)
+}
+
+type sourcesClientImpl struct {
+	client ClientWithResponsesInterface
+}
+
+func createGetSourcesResponseList(resp []Source) []SourceData {
+	sourceResponseList := make([]SourceData, len(resp))
+	for i, response := range resp {
+		sourceResponseList[i] = SourceData{
+			ID:                 string(*response.Id),
+			Name:               *response.Name,
+			AvailabilityStatus: *response.AvailabilityStatus,
+		}
+	}
+
+	return sourceResponseList
+}
+
+func createGetRhcConnectionStatus(resp RhcConnectionRead) RhcConnectionStatus {
+	rhcConnectionStatus := RhcConnectionStatus{
+		ID:                 string(*resp.Id),
+		RhcID:              *resp.RhcId,
```

Comment:
You might want to watch out for nils in this code.  I think it might be possible for a source to not have an rhc_id.  In that case, this code would cause a panic.

Here's an example test case:

```
		It("Errors when Source does not have an rhc_id", func() {
			doer := test.MockHttpClient(200, `{"id": "1", "availability_status": "connected"}`)

			client := NewSourcesClientWithHttpRequestDoer(config.Get(), &doer)
			ctx := test.TestContext()
			sourceId := ID("12345")
			result, err := client.GetRHCConnectionStatus(ctx, sourceId)
			Expect(err).ToNot(HaveOccurred())
			Expect(result).To(Equal(RhcConnectionStatus{
				ID:                 "1",
				AvailabilityStatus: "connected",
			}))

		})
```

Reply:
will do!